arclightning_backend/src/main.rs

```diff
@@ -0,0 +1,108 @@
+extern crate futures;
+extern crate hyper;
+#[macro_use] extern crate serde_derive;
+#[macro_use] extern crate serde_json;
+extern crate toml;
+
+use futures::future;
+use hyper::rt::{Future, Stream};
+use hyper::service::service_fn;
+use hyper::{Body, Method, Request, Response, Server, StatusCode};
+use std::path::PathBuf;
+
+// after this concept is further understood, will switch to 'Either'
+type BoxFuture = Box<Future<Item = Response<Body>, Error = hyper::Error> + Send>;
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
```

Comment:
I've taken to calling this type `ResponseFuture` in personal code. I think we can do that here, and it would improve readability a bit.